Source/SwiftLintBuiltInRules/Rules/Metrics/NestingRule.swift

```diff
@@ -12,137 +13,171 @@ struct NestingRule: Rule {
+            validate(node, triggeringToken: node.funcKeyword, inFunction: validationData.isFunction)
+            return .visitChildren
+        }
+
+        override func visitPost(_ node: FunctionDeclSyntax) {
+            validationData.pop()
+        }
+
+        override func visit(_ node: StructDeclSyntax) -> SyntaxVisitorContinueKind {
+            validate(node, triggeringToken: node.structKeyword, inFunction: validationData.isFunction)
+            return .visitChildren
+        }
+
+        override func visitPost(_ node: StructDeclSyntax) {
+            validationData.pop()
+        }
+
+        // MARK: - configuration for ignoreTypealiasesAndAssociatedtypes
+        override func visitPost(_ node: TypeAliasDeclSyntax) {
+            guard !configuration.ignoreTypealiasesAndAssociatedtypes else { return }
+            validate(node, triggeringToken: node.typealiasKeyword, inFunction: false)
+            validationData.pop()
+        }
+
+        // associatedtype is only permitted within a protocol,
+        // so it maybe not necessary to check the configuration.
+        override func visitPost(_ node: AssociatedTypeDeclSyntax) {
+            guard !configuration.ignoreTypealiasesAndAssociatedtypes else { return }
+            validate(node, triggeringToken: node.associatedtypeKeyword, inFunction: false)
+            validationData.pop()
+        }
+
+        // MARK: - configuration for checkNestingInClosuresAndStatements
+        override func visit(_ node: ClosureExprSyntax) -> SyntaxVisitorContinueKind {
+            if !configuration.checkNestingInClosuresAndStatements {
+                return .skipChildren
+            } else {
+                return super.visit(node)
+            }
         }
 
-        guard let severity = violatingSeverity else {
-            return nil
+        override func visit(_ node: CodeBlockItemSyntax) -> SyntaxVisitorContinueKind {
+            if !configuration.checkNestingInClosuresAndStatements && node.parent?.inStatement ?? false {
+                return .skipChildren
+            } else {
+                return super.visit(node)
+            }
         }
 
-        let targetName = forFunction ? "Functions" : "Types"
-        let threshold = configuration.threshold(with: targetLevel, for: severity)
-        let pluralSuffix = threshold > 1 ? "s" : ""
-        return StyleViolation(
-            ruleDescription: Self.description,
-            severity: severity,
-            location: Location(file: file, byteOffset: offset),
-            reason: "\(targetName) should be nested at most \(threshold) level\(pluralSuffix) deep"
-        )
+        // MARK: -
+        private func validate(_ node: some DeclSyntaxProtocol, triggeringToken: TokenSyntax? = nil, inFunction: Bool) {
+            validationData.push(node)
+            let isFunction = validationData.isFunction
+            let (level, targetLevel) = if isFunction {
```

Comment:
While I'm not agains the new `if` expressions, I'm still unsure how to format them in a clear and readable way. Either

```swift
let x = if ... {
            a
        } else {
            b
        }
```
or
```swift
let x =
    if ... {
        a
    } else {
        b
    }
```
look a bit better, though.

Reply:
Thanks!, The former alignment is too much work, so I changed to the latter.